### PR TITLE
[mailio] correct hint

### LINKED ISSUE #23798: [mailio] hint message after installing mailio is incorrect
After installing mailio, the hint message is wrong:

```
...
The package mailio provides CMake targets:

    find_package(cmake CONFIG REQUIRED)
    target_link_libraries(main PRIVATE mailio)
...
```

cmake, in find_package line, should be mailio.

### PR DESCRIPTION
**Pull request by david-antiteum**
- #### What does your PR fix?
  Fixes #23798

### CODE REVIEW

**Comment by dg0yt** [comment]
Could the mailio cmake config be found at all, before this change?

File: ports/mailio/portfile.cmake [hunk 1]
```diff
@@ -17,6 +17,11 @@ vcpkg_cmake_configure(
...
 vcpkg_cmake_install()
 
+vcpkg_fixup_cmake_targets(
+     CONFIG_PATH share/mailio/cmake
+     TARGET_PATH share/mailio
+)
+
 vcpkg_fixup_pkgconfig()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```
Comment (LilyWangLL) [inline ports/mailio/portfile.cmake:27]: `file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")` can be removed, it has been fixed in `vcpkg_fixup_cmake_targets`.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
Thanks for your PR, `string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" BUILD_SHARED)` should be removed, I can found where `BUILD_SHARED` is used.

File: ports/mailio/portfile.cmake [hunk 2]
```diff
@@ -17,6 +17,11 @@ vcpkg_cmake_configure(
 )
 vcpkg_cmake_install()
 
+vcpkg_fixup_cmake_targets(
```
Comment (dg0yt) [inline ports/mailio/portfile.cmake:20]: This legacy function has been replaced by `vcpkg_cmake_config_fixup` (from port vcpkg-cmake-config).

Comment (LilyWangLL) [reply ports/mailio/portfile.cmake:20]: It also need add dependency `vcpkg-cmake-config` in **vcpkg.json**.

**Comment by david-antiteum** [comment]
@LilyWangLL I just saw this MR #23772 that contain some of these fixes. Maybe you can accept first that MR and then update this one if something is missing.

**Comment by LilyWangLL** [comment]
I think PR #23772 can closed and its fix can merged to this PR, it just add `vcpkg_cmake_config_fixup`. So could you address the review suggestions? Thanks.

**Comment by david-antiteum** [comment]
Done.

**Comment by LilyWangLL** [comment]
Hi @david-antiteum, can you please resolve the conflicts against master?

**Comment by LilyWangLL** [comment]
@david-antiteum Please update `port-version` in `vcpkg.json`. Your version change was overwritten by #23772.

**Review by LilyWangLL — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**